incubator/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/stream/KafkaGrpcRemoteServerFactory.java

```diff
@@ -156,18 +160,21 @@ public void attach(
         KafkaGrpcBindingConfig newBinding = new KafkaGrpcBindingConfig(binding);
         bindings.put(binding.id, newBinding);
 
-        newBinding.routes.forEach(r ->
-            r.when.forEach(c ->
-            {
-                KafkaGrpcConditionResult condition = c.resolve();
-                servers.add(
-                    new KafkaRemoteServer(newBinding.id, newBinding.options.entryId, r.id, condition, newBinding.helper));
-            }));
-
-        this.reconnectAt = signaler.signalAt(
-            currentTimeMillis(),
-            SIGNAL_INITIATE_KAFKA_STREAM,
-            this::onKafkaStreamInitializationSignal);
+        if (canInitiate.apply(binding.id))
```

Comment:
We need a similar flow in `detach` to decrement the counter and remove it from the map when it reaches `0`.

This will prevent a potential memory leak in the concurrent map.

Perhaps rename `canInitiate` to `activate`, and add `deactivate`.